src/nvim/search.c

```diff
@@ -1527,7 +1527,7 @@ int searchc(cmdarg_T *cap, int t_cmd)
     if (*lastc == NUL && lastc_bytelen == 1) {
       return FAIL;
     }
-    if (dir) {        // repeat in opposite direction
+    if (dir != 0) {        // repeat in opposite direction
```

Comment:
Given

```c
/// Directions.
typedef enum {
  kDirectionNotSet = 0,
  FORWARD = 1,
  BACKWARD = (-1),
  FORWARD_FILE = 3,
  BACKWARD_FILE = (-3),
} Direction;
```

shouldn't this be

```c
if (dir < 0) {
```

Reply:
Hmm, but upstream is the same...

Reply:
Ah, this is specific to the handling of `;` and `,`, which use `true`/`false` for `eap->arg` instead of the Direction enum.

Reply:
Should probably change `Direction dir = cap->arg;` to `int dir = cap->arg;`.

Reply:
I've changed variable type "direction" to "int". Also reverted the change with dir since the warning shouldn't show up for ints.